example-schemas/swagger-petstore/scripts/convert-petstore.ts

```diff
@@ -0,0 +1,476 @@
+): ir.Type {
+  if (oasProperty.$ref) {
+    const typePath = schemaRefToTypePath(oasProperty.$ref);
+    return { type: "Plain", valueTypePath: typePath };
+  }
+  if (!("type" in oasProperty)) {
+    return { type: "Plain", valueTypePath: "unknown" };
+  }
+  if (oasProperty.type === "array") {
+    const items = oasProperty.items as oas.Referenced<oas.Oas3_1Schema>;
+    const itemType = getFieldType(items);
+    return { type: "Array", valueTypePath: itemType.valueTypePath };
+  }
+  if (oasProperty.type === "string") {
+    if (oasProperty.enum) {
+      const matchedEnum = Object.entries(result.defs).filter(([, def]) =>
+        def.type === "Enum"
+      ).find(([, value]) => {
+        const items = (value as ir.Enum).items.map(({ name }) => name);
+        return (oasProperty.enum as string[]).every((item) =>
+          items.includes(item)
+        );
+      });
+      if (matchedEnum) return { type: "Plain", valueTypePath: matchedEnum[0] };
+    }
+    if (oasProperty.format) {
+      if (["date", "date-time"].includes(oasProperty.format)) {
+        return {
+          type: "Plain",
+          valueTypePath: oasProperty.format.replace("-", ""),
+        };
+      } else {
+        console.log("unexpected string format", oasProperty.format);
+      }
+    }
+    return { type: "Plain", valueTypePath: "string" };
+  }
+  if (oasProperty.type === "integer") {
+    const valueTypePath = oasProperty.format || "integer";
+    return { type: "Plain", valueTypePath };
+  }
+  if (oasProperty.type === "number") {
+    const valueTypePath =
+      (oasProperty.format === "double" ? "float64" : oasProperty.format) ||
+      "float64";
+    return { type: "Plain", valueTypePath };
+  }
+  if (oasProperty.type === "boolean") {
+    return { type: "Plain", valueTypePath: "boolean" };
+  }
+  if (oasProperty.type === "object") {
+    if ("properties" in oasProperty) {
+      console.log("unexpected object property", oasProperty);
+    }
+    return { type: "Plain", valueTypePath: "object" };
+  }
+  console.log("unexpected property type");
+  return { type: "Plain", valueTypePath: "unknown" };
+}
+
+function buildOperation(
+  httpPath: string,
+  httpMethod: string,
+  operation: oas.Oas3Operation,
+): void {
+  const name = camelToPascal(operation.operationId || "");
+  const modulePath = tagToModulePath(operation.tags?.[0] || "data");
+  const defPath = `${modulePath}.${name}`;
+  const proc: ir.Proc = {
+    type: "Proc",
+    attributes: {},
+    name,
+    inputType: voidType,
+    outputType: voidType,
+  };
+  proc.attributes.http = `${httpMethod.toUpperCase()} ${httpPath}`;
+  if (operation.security) {
+    const items: string[] = [];
+    for (const requirement of operation.security) {
+      for (const [key, value] of Object.entries(requirement)) {
+        if (value.length) items.push(`${key}: ${value.join(", ")}`);
+        else items.push(key);
+      }
+    }
+    proc.attributes.security = items.join("\n");
+  }
```

Comment:
Let's just stringify this part as YAML and insert it after trimming.

Reply:
```suggestion
  if (operation.security) {
    proc.attributes.security = stringify(operation.security);
  }
```
did you mean output like this? it seems little bit odd to me when the item is single.
```
// as-is
@ security - petstore_auth: write:pets, read:pets
// to-be
@ security
| - petstore_auth:
|     - 'write:pets'
|     - 'read:pets'
```
```
// as-is
@ security - api_key
// to-be
@ security - - api_key: []
```

Reply:
I agree that the expression I suggested sounds awkward.
But I'm not sure if the abbreviated expression properly covers the existing usages.